### PR TITLE
Add metrics to LORA pass

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add metrics to LORA pass. Only AML system is supported now.

e.g.
image

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/passes/pytorch/lora.py [hunk 1]
```diff
@@ -620,6 +625,23 @@ def train_and_save_new_model(
             # train
             logger.info("Running fine-tuning")
             train_result = trainer.train()
+
+            # log metrics now only support Azure ML system. Metrics can be found in Metrics tab in Azure ML job
```
Comment (jambayk) [inline olive/passes/pytorch/lora.py:629]: I don't think manual logging of metrics might be needed. Can you try using `"report_to": "mlflow"` under the training arguments config. 
That should directly log the metrics to mflow in realtime. https://huggingface.co/docs/transformers/v4.38.1/en/main_classes/trainer#transformers.TrainingArguments.report_to

File: olive/passes/pytorch/lora.py [hunk 2]
```diff
@@ -620,6 +625,23 @@ def train_and_save_new_model(
...
+            if config.log_metrics:
+                try:
+                    import mlflow
+
+                    logs = trainer.state.log_history
+                    logger.debug(f"trainer log history: {logs}")
```
Comment (guotuofeng) [inline olive/passes/pytorch/lora.py:635]: please rebase from main which disable the f-string logging.

File: olive/passes/pytorch/lora.py (hunk 1 above)
Comment (xiaoyu-work) [reply olive/passes/pytorch/lora.py:629]: Wow good to know this! I tested and it worked. I'll close this PR. thanks!

Comment (jambayk) [reply olive/passes/pytorch/lora.py:629]: Awesome! thanks for checking

**Pull request closed without merging**